src/frontend/screens/Library/index.tsx

```diff
@@ -552,90 +553,7 @@ export default React.memo(function Library(): JSX.Element {
-                selected={selectedFilter}
-                targetWidth={275}
-              />
-            </div>
-            <div>
-              <GenericDropdown
-                target={
-                  <GenericDropdown.GenericButton
-                    text={'Other filters'}
-                    style={{ width: '340px' }}
-                  ></GenericDropdown.GenericButton>
-                }
-              >
-                {otherFiltersData.map((val, index) => (
-                  <Menu.Item
-                    closeMenuOnClick={false}
-                    key={`toggleItem${index}`}
-                  >
-                    <Toggle
-                      defaultChecked={val.defaultValue}
-                      labelPosition="right"
-                      onChange={
-                        //eslint-disable-next-line
-                        (e: any) => {
-                          val.onChange(e.target.checked)
-                        }
-                      }
-                    >
-                      <div
-                        className="body"
-                        style={{
-                          paddingLeft: 'var(--space-sm)',
-                          margin: 'auto 0px'
-                        }}
-                      >
-                        {val.text}
-                      </div>
-                    </Toggle>
-                  </Menu.Item>
-                ))}
-              </GenericDropdown>
-            </div>
-            <Tabs.Tab value="all">
-              <div className="menu">{t('ALL', 'ALL')}</div>
-            </Tabs.Tab>
-            <Tabs.Tab value="hyperplay">
-              <div className="menu">{t('HyperPlay')}</div>
-            </Tabs.Tab>
-            {isEpicLoggedin && (
-              <Tabs.Tab value="legendary">
-                <div className="menu">EPIC</div>
-              </Tabs.Tab>
-            )}
-            {isGOGLoggedin && (
-              <Tabs.Tab value="gog">
-                <div className="menu">GOG</div>
-              </Tabs.Tab>
-            )}
-            <Tabs.Tab value="sideload">
-              <div className="menu">{t('Other')}</div>
-            </Tabs.Tab>
-            <div id="alignEnd">
-              {/* <div>
-              <Button type="tertiary" className={styles.gridListButton}>
-                <Images.Grid fill="white" height={24} width={24} />
-              </Button>
-            </div>
-            <div>
-              <Button type="tertiary" className={styles.gridListButton}>
-                <Images.List fill="white" height={24} width={24} />
-              </Button>
-            </div> */}
-            </div>
-          </Tabs.List>
-        </Tabs>
+        <LibraryTopBar handleCategory={handleCategory} category={category} filters={filters} setSelectedFilter={setSelectedFilter} selectedFilter={selectedFilter} otherFiltersData={otherFiltersData} isEpicLoggedin={isEpicLoggedin} isGOGLoggedin={isGOGLoggedin}/>
```

Comment:
`isEpicLoggedin`, `isGOGLoggedin` can be calculated in 
 `<LibraryTopBar/>` via `useContext(ContextProvider)`.

`category` and `handleCategory` are available through the context too.

Overall, I think that you can use a context or a store to handle the state and avoid prop drilling for this refactor

Reply:
I appreciate refactor initiative btw 🔥

Reply:
tru. after the refactor there are only a few props though and it is only one level. i think it is fine to use props for that